src/PKSim.Core/Services/HIDiseaseStateImplementation.cs

```diff
@@ -0,0 +1,232 @@
+      private static readonly HIFactors _otherOrgansFlowScalingFactor = createFactors(1.31, 1.84, 2.27);
+      private static readonly HIFactors _gfrScalingFactor = createFactors(1, 0.7, 0.36);
+      private static readonly HIFactors _albuminFactor = createFactors(0.81, 0.68, 0.5);
+      private static readonly HIFactors _agpFactor = createFactors(0.6, 0.56, 0.3);
+      private static readonly HIFactors _hematocritScalingFactor = createFactors(0.92, 0.88, 0.83);
+
+      private static readonly Cache<string, HIFactors> _moleculeScalingFactorEdginton = new()
+      {
+         {"CYP3A4", createFactors(1, 0.4, 0.4)},
+         {"CYP1A2", createFactors(1, 0.1, 0.1)},
+         {"CYP2E1", createFactors(1, 0.83, 0.83)},
+      };
+
+      private static readonly Cache<string, HIFactors> _moleculeScalingFactorJohnson = new()
+      {
+         {"CYP2A6", createFactors(0.89, 0.62, 0.32)},
+         {"CYP2B6", createFactors(1, 0.9, 0.8)},
+         {"CYP2C8", createFactors(0.69, 0.52, 0.33)},
+         {"CYP2C9", createFactors(0.69, 0.51, 0.33)},
+         {"CYP2C18", createFactors(0.32, 0.26, 0.12)},
+         {"CYP2C19", createFactors(0.32, 0.26, 0.12)},
+         {"CYP2D6", createFactors(0.76, 0.33, 0.11)},
+      };
+
+      public const string CHILD_PUGH_SCORE = "Child-Pugh score";
+
+      private const int HI_EDGINTON_VALUE_ORIGIN_ID = 93;
+      private const int HI_JOHNSON_VALUE_ORIGIN_ID = 94;
+
+      private readonly IDimension _ageDimension;
+
+      public HIDiseaseStateImplementation(IDimensionRepository dimensionRepository,
+         IParameterFactory parameterFactory,
+         IValueOriginRepository valueOriginRepository,
+         IFormulaFactory formulaFactory,
+         IIndividualFactory individualFactory,
+         IContainerTask containerTask,
+         IParameterSetUpdater parameterSetUpdater) : base(valueOriginRepository, formulaFactory, individualFactory, containerTask, parameterSetUpdater, CoreConstants.DiseaseStates.HI)
+      {
+         _parameterFactory = parameterFactory;
+         _ageDimension = dimensionRepository.AgeInYears;
+      }
+
+      public override bool ApplyTo(Individual individual) => applyTo(individual, UpdateParameter(HI_EDGINTON_VALUE_ORIGIN_ID));
+
+      public override bool ApplyForPopulationTo(Individual individual) => applyTo(individual, UpdateParameterValue);
+
+      private bool applyTo(Individual individual, Action<IParameter, double, bool> updateParameterFunc)
+      {
+         updateBloodFlows(individual, updateParameterFunc);
+         updateVolumes(individual, updateParameterFunc);
+         updateGFR(individual, updateParameterFunc);
+         updateOntogenyFactory(individual, updateParameterFunc);
+         updateHematocrit(individual, updateParameterFunc);
+
+         addChildPughScoreToLiver(individual);
+         return true;
+      }
+
+      private void addChildPughScoreToLiver(Individual individual)
+      {
+         var organism = individual.Organism;
```

Comment:
@Yuri05 I am adding this parameter under the organism as readonly so that we can differentiate between individuals in a population. Do you see a problem? It's a readonly constant value that comes from the origin data

Reply:
Don't see a problem right now. We will see :)